testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -387,12 +389,81 @@ public void testUpdateAccessControl()
     @Test
     public void testMergeAccessControl()
     {
-        assertAccessAllowed("""
-                MERGE INTO orders o USING region r ON (o.orderkey = r.regionkey)
-                WHEN MATCHED AND o.orderkey % 2 = 0 THEN DELETE
-                WHEN MATCHED AND o.orderkey % 2 = 1 THEN UPDATE SET orderkey = null
-                WHEN NOT MATCHED THEN INSERT VALUES (null, null, null, null, null, null, null, null, null)
-                """);
+        String catalogName = getSession().getCatalog().orElseThrow();
+        String schemaName = getSession().getSchema().orElseThrow();
+
+        String targetTable = "merge_nation_target_" + randomTableSuffix();
+        String targetName = format("%s.%s.%s", catalogName, schemaName, targetTable);
+        String sourceTable = "merge_nation_source_" + randomTableSuffix();
+        String sourceName = format("%s.%s.%s", catalogName, schemaName, sourceTable);
+
+        assertUpdate(format("CREATE TABLE %s (nation_name VARCHAR, region_name VARCHAR)", targetTable));
+
+        assertUpdate(format("INSERT INTO %s (nation_name, region_name) VALUES ('FRANCE', 'EUROPE'), ('ALGERIA', 'AFRICA'), ('GERMANY', 'EUROPE')", targetTable), 3);
```

Comment:
No need to insert data, since we aren't testing the actual merge functionality, and the blackhole connector will discard the data anyway. (so it's confusing to the reader as to why we do it)

Reply:
Removed.